lib/Pingmachine/Probe/HTTPing.pm

```diff
@@ -0,0 +1,242 @@
+            $self->_collect_current_job();
+
+            $log->debug("collected:". Dumper(\$job{cmd}) . "(step: $step, pings: $pings, offset: " . $self->time_offset() . ")") if $log->is_debug();
+        }
+    );
+}
+
+sub _kill_current_job {
+    my ($self) = @_;
+
+    # Kill httping, if still running
+    my $job = $self->current_job;
+    for my $url (keys %{$job->{url2order}}) {
+        if($job->{pid}{$url}) {
+            # Check that we are killing the process we started and not an innocent bystander
+            my $cmd_match = 0;
+            if (open(my $proc_fh, "/proc/$job->{pid}{$url}/cmdline")) {
+                $cmd_match = (join('', readline($proc_fh)) eq $job->{cmd}{$url});
+                close($proc_fh);
+            }
+            if($cmd_match && kill(0, $job->{pid}{$url})) {
+                $log->warning("killing unfinished httping process (step: ".$self->step.", pings: ".$self->pings.", offset: ".$self->time_offset().")");
+                kill 9, $job->{pid}{$url};
+                $job->{pid}{$url} = undef;
+            }
+            elsif($job->{output}{$url}) {
+                $log->warning("httping has finished, but we didn't notice... collecting (step: ".$self->step.", pings: ".$self->pings.", offset: ".$self->time_offset.")");
+                $self->_collect_current_job();
+            }
+            else {
+                $log->warning("httping has finished, but we didn't notice... no output found (?)");
+            }
+        }
+    }
+}
+
+sub _collect_current_job {
+    my ($self) = @_;
+    my $text;
+
+    my $job = $self->current_job;
+    $self->current_job({});
+    my %results;
+
+    # Parse httping report
+    for my $url (keys %{$job->{url2order}}) {
+        # Do nothing, if httping didn't run yet or if job has been already collected
+        return unless $job->{output}{$url};
+        
+        my $raw_text = $job->{output}{$url};
+        # sample output
+        # PING neverssl.com:80 (/):
+        #	connected to neverssl.com:80 (524 bytes), seq=0 time= 10.86 ms
+        #	connected to neverssl.com:80 (524 bytes), seq=1 time= 15.46 ms
+        #	--- http://neverssl.com/ ping statistics ---
+        #	2 connects, 2 ok, 0.00% failed, time 2027ms
+        #	round-trip min/avg/max = 10.9/13.2/15.5 ms
+        # parse line by line httping output and format it as with fping
+        my @lines = split /\n/, $raw_text;
+        my @data;
+        for my $line (@lines) {
+            # if the line contains an error (i.e. timeout, connection refused, ...) add a -
+            if ($line =~ /could not connect|timeout|short read/) {
+                push @data, '-';
+            }
+            # if the line contains the httping result add the number
+            if ($line =~ /time=\s*(\d\d*.\d\d*) ms/) {
+                push @data, $1;
+            }
+            # ignore all the rest
+        }
+        # raw ping times
+        my @pings = map {$_ eq '-' ? undef : $_ / 1000} @data;
```

Comment:
I suggest to to rename `@data` to `@pings` and put them in already in the form used later on.
That would then also simplify the generation of `@rtts` to
```
my @rtts = map {sprintf "%.6e", $_ } sort {$a <=> $b} grep {$_} @pings;
```